frontend/src/authentication/components/AuthGuard.js

```diff
@@ -1,18 +1,58 @@
 import PropTypes from 'prop-types';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Navigate, useLocation } from 'react-router-dom';
 import { Login } from '../views/Login';
 import { useAuth } from '../hooks';
 import {
   RegexToValidateWindowPathName,
   WindowPathLengthThreshold
 } from '../../utils';
+import { useClient, useGroups } from '../../services';
+import { LoadingScreen, NoAccessMaintenanceWindow } from '../../design';
+import { getMaintenanceStatus } from '../../services/graphql/MaintenanceWindow';
+import {
+  ACTIVE_STATUS,
+  PENDING_STATUS
+} from '../../modules/Administration/components';
+import config from '../../generated/config.json';
+import { SET_ERROR, useDispatch } from '../../globalErrors';
 
 export const AuthGuard = (props) => {
   const { children } = props;
   const auth = useAuth();
   const location = useLocation();
   const [requestedLocation, setRequestedLocation] = useState(null);
+  const [isNoAccessMaintenance, setNoAccessMaintenanceFlag] = useState(null);
+  const client = useClient();
+  const groups = useGroups();
+  const dispatch = useDispatch();
+
+  const checkMaintenanceMode = async () => {
+    const response = await client.query(getMaintenanceStatus());
+    if (!response.errors && response.data.getMaintenanceWindowStatus != null) {
+      if (
+        [PENDING_STATUS, ACTIVE_STATUS].includes(
+          response.data.getMaintenanceWindowStatus.status
+        ) &&
+        response.data.getMaintenanceWindowStatus.mode === 'NO-ACCESS' &&
+        !groups.includes('DAAdministrators')
+      ) {
+        setNoAccessMaintenanceFlag(true);
+      } else {
+        setNoAccessMaintenanceFlag(false);
+      }
+    }
+  };
```

Comment:
Suggestion, can we use a try/catch block here like the following:

<img width="843" alt="Screenshot 2024-05-03 at 4 48 52 PM" src="https://github.com/data-dot-all/dataall/assets/97474536/1534dfe5-c67b-487d-b35b-1e0713cc9167">

`
const checkMaintenanceMode = async () => {
    try {
        const response = await client.query(getMaintenanceStatus());
        const maintenanceStatus = response?.data?.getMaintenanceWindowStatus;

        if (maintenanceStatus && [PENDING_STATUS, ACTIVE_STATUS].includes(maintenanceStatus.status) && maintenanceStatus.mode === 'NO-ACCESS' && !groups.includes('DAAdministrators')) {
            setNoAccessMaintenanceFlag(true);
        } else {
            setNoAccessMaintenanceFlag(false);
        }
    } catch (error) {
        // Handle error
    }
};

`

Reply:
@rbernotas, for this async method, the errors from client.query are attached in the `response`. This doesn't throw an exception. 

For catching unexpected exception , this function is catching exception in `catch` block defined in the useEffect hook. 
```js
if (config.modules.maintenance.active === true) {
      if (client) {
        checkMaintenanceMode().catch((e) => dispatch({ type: SET_ERROR, e }));
      }
}
```